akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -950,6 +959,9 @@ class PersistentShardCoordinator(
 
   def waitingForStateInitialized: Receive =
```

Comment:
There are several `Receive` cases in `DDataShardCoordinator`, which might also have to handle Terminate. Most of them are stashing so it will be handled after unbecome

In `waitingForUpdate` I think it can stop immediately because that is during the startup phase.

`waitingForUpdate` I think it needs to keep track of if it has received Terminate and stop after UpdateTimeout, otherwise it may repeat the retries forever (until majority write is successful). Maybe it's a more safe option to actually not stop it if those updates fail, but I'm somewhat uncertain if we should limit the number of retries for this stopping scenario.

If you aren not familiar with Distributed Data I could follow up with the changes needed there.

Reply:
I am not very familiar with DData, so any help/advise is welcome. I added handling of Terminate command in DdataShardCoordinator as well, basically approach is as follows, actor is stopped if state is not yet collected. If state is collected and some update is pending when Terminate is received I stash that message and mark ShardCoordinator state as terminated. Upon receiving either XxxTimeout or ModifyFailure i call for `unbecome` and `unstashAll`.